### PR TITLE
Adding django-rest-framework-filters for NOT filters

### PR DESCRIPTION
**Pull request by daviddavis**
https://github.com/philipn/django-rest-framework-filters

fixes #5868
https://pulp.plan.io/issues/5868

Please be sure you have read our documentation on creating PRs:
https://docs.pulpproject.org/en/3.0/nightly/contributing/pull-request-walkthrough.html

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
     'backoff',
     'coreapi~=2.3.3',
     'Django~=2.2.3',  # LTS version, switch only if we have a compelling reason to
-    'django-filter~=2.2.0',
+    'djangorestframework-filters==1.0.0.dev0',
```
Comment (daviddavis) [inline setup.py]: I tried using `>=` here but it didn't work (it installed 0.11.1). 

It looks [doubtful about whether we'll see a 1.0 release](https://github.com/philipn/django-rest-framework-filters/issues/287) which makes me nervous.

**Comment by daviddavis** [comment]
Waiting on a response to https://github.com/philipn/django-rest-framework-filters/issues/287#issuecomment-574312703

**Comment by dralley** [comment]
@daviddavis How easily could the not-equal filter be vendored?

**Comment by daviddavis** [comment]
@dralley I don't think it would be too hard although I haven't vendored anything in Python before. I could look into it.

My concern though is having to maintain the django-rest-framework-filters code. Like right now, [it looks like it doesn't support django 3](https://github.com/philipn/django-rest-framework-filters/issues/326).

**Comment by daviddavis** [comment]
@dralley brought up a good point. We can't name fields with `!` since these field names are used in the bindings and `!` is a special character in many languages (including Python).

**Pull request closed without merging**